tools/export/exporters.py

```diff
@@ -130,9 +130,13 @@ def make_key(self, src):
         Positional Arguments:
         src - the src's location
         """
-        key = basename(dirname(src))
-        if key == ".":
-            key = basename(realpath(self.export_dir))
+        rel_path = relpath(src, self.resources.file_basepath[src])
+        path_list = os.path.normpath(rel_path).split(os.sep)
+        assert path_list >= 1
```

Comment:
Is that correct behavior though? say src is ```C:\Project\mbed-os\src.cpp```. Then ```relpath=mbed-os\src.cpp```, then ```path_list = [mbed-os, src.cpp]```, which would be a list length >=1.